kad.py

```diff
@@ -34,39 +34,54 @@ def get(self, config_name, valuse_name):
 
 class Krita:
     __base_url__ = "https://binary-factory.kde.org/job/Krita_Nightly_Android_Build/"
+    __latest__ = False
 
     app_name = 'org.krita'
     required_space = 1.5
 
     def __init__(self, file_path, version=''):
         if not version:
-            Console.display_message('Checking new version')
-
-            self.__build_version__ = 'undefined'
-
-            res = get(self.__base_url__, verify=False)
-
-            if res.status_code is not 200:
-                Console.display_error(f'Version check failed, response statue: {res.status_code}')
-
-            res = str(res.content).split("\\n")
-            for item in res:
-                if 'Last stable build' in item:
-                    self.__build_version__ = item.split('(')[1].split(')')[0].replace('#', '')
-                    Console.display_message(f'Latest version: {self.__build_version__}')
+            self.__build_version__ = self.__get_latest_version_number__()
+            self.__latest__ = True
         else:
             self.__build_version__ = version
+            self.__latest__ = False
 
         if not file_path[-1] is '/' and file_path[-1] is not '\\':
             file_path += '/'
 
         self.file_url = f'{file_path}krita_build_apk-release-{self.__build_version__}-unsigned.apk'
         self.__download_url__ = f'{self.__build_version__}/artifact/krita_build_apk-release-unsigned.apk'
 
+    def __get_latest_version_number__(self):
```

Comment:
No return value if version page does not contain a line with 'Last stable build'